src/main/java/org/broadinstitute/dsm/model/BSPKit.java

```diff
@@ -0,0 +1,150 @@
+package org.broadinstitute.dsm.model;
+
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.InstanceSettings;
+import org.broadinstitute.dsm.db.dao.ddp.kitrequest.KitRequestDao;
+import org.broadinstitute.dsm.db.dto.settings.InstanceSettingsDto;
+import org.broadinstitute.dsm.model.bsp.BSPKitInfo;
+import org.broadinstitute.dsm.model.bsp.BSPKitQueryResult;
+import org.broadinstitute.dsm.model.bsp.BSPKitStatus;
+import org.broadinstitute.dsm.statics.ApplicationConfigConstants;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BSPKit {
+    private static Logger logger = LoggerFactory.getLogger(BSPKit.class);
+
+    public BSPKitStatus getKitStatus(@NonNull String kitLabel, NotificationUtil notificationUtil) {
+        logger.info("Checking label " + kitLabel);
+        BSPKitQueryResult bspKitQueryResult = BSPKitQueryResult.getBSPKitQueryResult(kitLabel);
```

Comment:
`BSPKitQueryResult` class can be easily separated into `Dao/Dto` since it has only one method `getBSPKitQueryResult`, Optional can be applied on  `getBSPKitQueryResult` as well, which will help you to make later if check more cleaner.